kernel/src/syscall_driver.rs

```diff
@@ -209,6 +214,24 @@ pub trait SyscallDriver {
         Err((buffer, ErrorCode::NOSUPPORT))
     }
 
+    /// System call for a process to pass a buffer (a UserspaceReadableProcessBuffer) to
+    /// the kernel that the kernel can either read or write. The kernel calls
+    /// this method only after it checks that the entire buffer is
+    /// within memory the process can both read and write.
+    ///
+    /// This is different to `allow_readwrite()` in that the app is allowed
+    /// to read the buffer once it has been passed to the kernel.
+    /// For more details on how this can be done safely see the userspace
+    /// readable allow syscalls TRD.
```

Comment:
link? or at least TRD number (can XXX for now)?

Reply:
Ha we went around on this already, I think this is fine until there is actually a number.